Domain/Services/BookingService.cs

```diff
@@ -0,0 +1,115 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+using AirportTicketBookingSystem.Domain.Models;
+using AirportTicketBookingSystem.Domain.Repositories;
+
+namespace AirportTicketBookingSystem.Domain.Services
+{
+    public class BookingService
+    {
+        private readonly BookingRepository _bookingRepository;
+        private readonly FlightService _Flights;
+
+        public BookingService()
+        {
+            _bookingRepository = new BookingRepository();
+            _Flights = new FlightService();
+        }
+
+        public void AddBooking(Booking booking)
+        {
+            _bookingRepository.AddBooking(booking);
+            
+        }
+
+        public List<Booking> GetBookingsByPassenger(int passengerId)
+        {
+            return _bookingRepository.GetBookingsByPassenger(passengerId);
+        }
+
+        public void CancelBooking(int bookingId)
+        {
+            _bookingRepository.CancelBooking(bookingId);
+            
+        }
+
+        public List<Booking> GetAllBookings()
+        {
+            return _bookingRepository.GetAllBookings();
+        }
+
+        public Booking GetLastBooking()
+        {
+            return _bookingRepository.GetLastBooking();
+        }
+
+        public void ModifyBooking(int bookingId, int newFlightId, string newSeatClass)
+        {
+            var booking = _bookingRepository.GetBookingById(bookingId);
+            if (booking == null)
+            {
+                Console.WriteLine("Booking not found.");
+                return;
+            }
+
+            booking.FlightId = newFlightId;
+            booking.SeatClass = newSeatClass;
+            _bookingRepository.UpdateBooking(booking);
+
+            Console.WriteLine($"Booking ID {bookingId} has been updated successfully.");
+        }
+
+
+        public List<Booking> FilterBookings(
+        double? price = null,
+        string departureCountry = null,
+        string destinationCountry = null,
+        DateTime? departureDate = null,
+        string departureAirport = null,
+        string arrivalAirport = null,
+        string passenger = null,
+        string seatClass = null)
+        {
+            List<Flight> filteredFlights = _Flights.SearchFlights(
+                departureCountry: departureCountry,
+                destinationCountry: destinationCountry,
+                departureAirport: departureAirport,
+                arrivalAirport: arrivalAirport,
+                departureDate: departureDate,
+                seatClass: seatClass,
+                maxPrice: price
+            );
+
+            var bookings = _bookingRepository.GetAllBookings();
```

Comment:
we shouldn't be getting all bookings and filter them, the idea of search is to only get the wanted records by one query without the need to retrieve all data for faster retrieval operations.